pytest_mh/utils/coredumpd.py

```diff
@@ -0,0 +1,205 @@
+
+        # If the file is compressed, remove the extension
+        if not parts[-1].isnumeric():
+            parts.pop()
+
+        pid = parts[-2]
+        timestamp = parts[-1]
+
+        return (pid, timestamp)
+
+    def get_artifacts_list(self, host: MultihostHost, type: MultihostArtifactsType) -> set[str]:
+        """
+        Dump backtrace and other information from generated core files for easy access.
+
+        :param host: Host where the artifacts are being collected.
+        :type host: MultihostHost
+        :param type: Type of artifacts that are being collected.
+        :type type: MultihostArtifactsType
+        :return: List of artifacts to collect.
+        :rtype: set[str]
+        """
+        if self._corefiles is None:
+            self._corefiles = self.list_core_files()
+
+        if not self._corefiles:
+            return set()
+
+        # Parse PID and timestamp that we can use to get information for journal
+        for name in self._corefiles:
+            try:
+                pid, timestamp = self.parse_core_file_name(name)
+            except ValueError:
+                self.logger.warn(f"Invalid core file name: {name}")
+                continue
+
+            # Dump the information
+            self.host.ssh.run(
+                rf"""
+                journalctl --output=verbose          \
+                    'COREDUMP_PID={pid}'             \
+                    'COREDUMP_TIMESTAMP={timestamp}' \
+                    > '{self.path}/{name}.backtrace'
+                """,
+                log_level=SSHLog.Error,
+            )
+
+        return {self.path}
+
+    def pytest_report_teststatus(
+        self, report: pytest.CollectReport | pytest.TestReport, config: pytest.Config
+    ) -> tuple[str, str, str | tuple[str, dict[str, bool]]] | None:
+        """
+        Report core file found error if found and matches requested filter.
+
+        :param report: Pytest report
+        :type report: pytest.CollectReport | pytest.TestReport
+        :param config: Pytest config
+        :type config: pytest.Config
+        :return: Pytest test status
+        :rtype: tuple[str, str, str | tuple[str, dict[str, bool]]] | None
+        """
+        if report.when != "call":
+            return None
+
+        if self.mode == "ignore" or report.outcome == "skipped":
+            return None
+
+        self.logger.info("Checking for core files")
+
+        if self._corefiles is None:
+            self._corefiles = self.list_core_files()
+
+        if not self._corefiles:
+            return None
+
+        # Ignore if no core files matches the filter
+        if self.filter:
+            match = re.search(self.filter, "\n".join(self._corefiles))
+            if match is None:
+                return None
+
+        original_outcome = report.outcome
+
+        # Fail the test if fail mode is selected
+        if report.outcome == "passed" and self.mode == "fail":
```

Comment:
what is the reason to have `if report.outcome == "passed"` here?

Reply:
If the test passed, we want to switch it to fail, otherwise don't touch the outcome. In theory, the outcome here may only be  "passed" or "failed" so this check is redundant. But I added it for verbosity and future safety.